src/components/Page/components/Header/Header.tsx

```diff
@@ -135,3 +217,120 @@ export function Header({
     </div>
   );
 }
+
+function shouldShowIconOnly(
```

Comment:
This is probably the most dense part of the PR. As part of this PR the page primary action now accepts an icon. If using the current design language, the icon is simply ignored. If using the new design language and an icon is supplied, the icon is hidden when on large screen showing just the button text. And on small screen it shows an icon only button.